cf-agent/verify_services.c

```diff
@@ -193,35 +173,13 @@ static PromiseResult VerifyServices(EvalContext *ctx, Attributes a, const Promis
 /* Level                                                                     */
 /*****************************************************************************/
 
-static FnCall *DefaultServiceBundleCall(const Promise *pp, ServicePolicy service_policy)
+static FnCall *DefaultServiceBundleCall(const Promise *pp, const char *service_policy)
 {
     Rlist *args = NULL;
     FnCall *call = NULL;
 
-    switch (service_policy)
-    {
-    case SERVICE_POLICY_START:
-        RlistAppend(&args, pp->promiser, RVAL_TYPE_SCALAR);
-        RlistAppend(&args, "start", RVAL_TYPE_SCALAR);
-        break;
-
-    case SERVICE_POLICY_RESTART:
-        RlistAppend(&args, pp->promiser, RVAL_TYPE_SCALAR);
-        RlistAppend(&args, "restart", RVAL_TYPE_SCALAR);
-        break;
-
-    case SERVICE_POLICY_RELOAD:
-        RlistAppend(&args, pp->promiser, RVAL_TYPE_SCALAR);
-        RlistAppend(&args, "restart", RVAL_TYPE_SCALAR);
-        break;
-
-    case SERVICE_POLICY_STOP:
-    case SERVICE_POLICY_DISABLE:
-    default:
-        RlistAppend(&args, pp->promiser, RVAL_TYPE_SCALAR);
-        RlistAppend(&args, "stop", RVAL_TYPE_SCALAR);
-        break;
```

Comment:
Not sure, but it seems that the default case is never hit because the service_policy is the result of a ServicePolicyFromString() call and this functions returns SERVICE_POLICY_START if the string is null or if it doesn't find the string in the possible values.